tests/awakener.rs

```diff
@@ -0,0 +1,89 @@
+use std::thread;
+use std::time::Duration;
+
+use mio_st::event::{Event, EventedId, Ready};
+use mio_st::poll::Awakener;
+
+mod util;
+
+use self::util::{expect_events, init_with_poller};
+
+#[test]
+fn awakener() {
+    let (mut poller, mut events) = init_with_poller();
+
+    let event_id = EventedId(10);
+    let awakener = Awakener::new(&mut poller, event_id)
+        .expect("unable to create awakener");
+
+    let handle = thread::spawn(move || {
+        awakener.wake().expect("unable to wake");
+    });
+
+    expect_events(&mut poller, &mut events, vec![
+        Event::new(event_id, Ready::READABLE),
+    ]);
+
+    handle.join().unwrap();
+}
+
+#[test]
+fn awakener_try_clone() {
+    let (mut poller, mut events) = init_with_poller();
+
+    let event_id = EventedId(10);
+    let awakener1 = Awakener::new(&mut poller, event_id)
+        .expect("unable to create awakener");
+    let awakener2 = awakener1.try_clone()
+        .expect("unable to clone awakener");
+
```

Comment:
The same problem exists here. If the system is under load, this test can spuriously fail.
Fix by keeping the Awakener alive.